music/src/music_vae/model.ts

```diff
@@ -999,14 +999,16 @@ class MusicVAE {
    * conditioning.
    * @param extraControls (Optional) Additional control tensors to use as
    * conditioning.
+   * @param key (Optional) Key to use as conditioning, an integer between 0
+   * (C major / A minor) and 11 (B major / G# minor).
    *
    * @returns An array of interpolation `NoteSequence` objects, as described
    * above.
    */
   async interpolate(
       inputSequences: INoteSequence[], numInterps: number|number[],
       temperature?: number, chordProgression?: string[],
-      extraControls?: tf.Tensor2D) {
+      extraControls?: tf.Tensor2D, key?: number) {
```

Comment:
I'm a little confused about the use of `extraControls`, when combined with key. What are the conditioning signals that are typically passed into `extraControls`?

Reply:
Here it's general, but I've been using it to pass melody rhythm, shape, and register.

Reply:
I feel a little bit bad about making the API less clean @adarob; if this is really bothersome I could clean it up a bit here and have some higher level wrapper class for doing a bunch of the melody experiments we want to try.  But either way I'd have to expose extraControls (or something a lot like it) here...

Reply:
Ideally we would make all of these optional arguments part of an interface that can be more easily extended. For example, we would have something like:

interface ConditioningArgs {
  chordProgression?: string[];
  extraControls?: tf.Tensor2D;
  key?: number;
}

This would be a breaking change for apps using chordProgression or extraControls, but we can do a full version number bump. Thoughts @notwaldorf ?

Reply:
I'd be happy with a full version bump, I'd like to remove some old code from the visualizer while we're at it too. We've been afraid of major bumping for too long

Reply:
like i feel 1.x -> 2.x is really scary, but like 4.x -> 5.x is peanuts

Reply:
Okay, I'll try to clean up the conditioning code.

Reply:
What do you think of the changes I just made?

Reply:
This looks much better to me!!

Reply:
Agreed. This is looking good